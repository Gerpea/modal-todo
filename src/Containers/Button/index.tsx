import React from 'react';
import clsx from 'clsx';

import style from './styles.module.scss';

export interface Props {
  type?: 'primary' | 'success' | 'danger';
  className?: string;
  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
  children: React.ReactNode;
}

function Button({ type = 'primary', className, onClick, children }: Props): React.ReactElement {
  const styles = clsx(className, style.Button, type && style[`Button--${type}`]);

  return (
    <button className={styles} onClick={onClick}>
      {children}
    </button>
  );
}

export default Button;
