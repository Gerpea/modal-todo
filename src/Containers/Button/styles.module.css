.Button {
  display: flex;
  flex-direction: row;
  align-items: center;

  border-radius: 3px;
  padding: 8px 12px;

  border: none;
  outline: none;

  font-family: var(--font-main);
  font-weight: bold;
  font-size: 0.875rem;
  text-transform: lowercase;
}

.Button--success {
  color: var(--c-white);
  background-color: var(--c-green);
}

.Button--danger {
  color: var(--c-white);
  background-color: var(--c-red);
}
