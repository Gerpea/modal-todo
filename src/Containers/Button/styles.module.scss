@import '../../variables';

.Button {
  display: flex;
  flex-direction: row;
  align-items: center;

  border: 2px solid transparent;
  border-radius: 3px;
  padding: 6px 10px; /* normal padding - border width */

  outline: none;

  font-weight: bold;
  font-size: 0.875rem;
  line-height: 1;
  text-transform: lowercase;

  cursor: pointer;

  &:focus,
  &:hover {
    background-color: transparent;
  }
}

.Button--primary {
  color: $c-white;
  background-color: $c-blue;

  &:focus,
  &:hover {
    color: $c-blue;
    border-color: $c-blue;
  }
}

.Button--success {
  color: $c-white;
  background-color: $c-green;

  &:focus,
  &:hover {
    color: $c-green;
    border-color: $c-green;
  }
}

.Button--danger {
  color: $c-white;
  background-color: $c-red;

  &:focus,
  &:hover {
    color: $c-red;
    border-color: $c-red;
  }
}
