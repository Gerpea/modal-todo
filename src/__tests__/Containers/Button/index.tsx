import React from 'react';
import renderer from 'react-test-renderer';

import Button from 'Containers/Button';

describe('Button test', () => {
  it('Button success', () => {
    const tree = renderer
      .create(
        <Button type="success" onClick={() => void {}}>
          success
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button danger close', () => {
    const tree = renderer
      .create(
        <Button type="danger" onClick={() => void {}}>
          danger
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
