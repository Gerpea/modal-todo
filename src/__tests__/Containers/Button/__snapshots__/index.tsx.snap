// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Button danger close 1`] = `
<button
  className="Button Button--danger"
  onClick={[Function]}
>
  danger
</button>
`;

exports[`Button test Button success 1`] = `
<button
  className="Button Button--success"
  onClick={[Function]}
>
  success
</button>
`;
